### ISSUE TITLE
Investigate using crash Telemetry pings

### ISSUE THREAD

**Issue by Sdaswani** [mozilla-mobile/firefox-tv#1379]
**edit: As mentioned below, Glean is going to implement this so we may not want to do redundant work on the current telemetry implementation. Follow up on the email discussion before proceeding with this bug.**

---

Investigate option to send a crash ping, so that we can keep track of our crash rates. It will also help us compare between releases.

@pocmo said it best: https://github.com/mozilla-mobile/focus-android/issues/2348#issue-306818296

*From Focus issue:*
The crash ping will help us to measure the health of the app and hopefully avoid situations like we had in Focus where we couldn't really pinpoint certain issues.

The glean, app and AC teams have met and are going to work on a solution to collect this data with Glean from the crash reporter component.

In AC that is tracked by the meta "Good crash statistics":
[mozilla-mobile/android-components#1820](https://github.com/mozilla-mobile/android-components/issues/1820)

The related Glean issue is:
https://bugzilla.mozilla.org/show_bug.cgi?id=1522461

**Comment by mcomella** [comment]
A few things we should verify:
- Are crash pings the best solution to come up with crash rates?
- How will our crash ping interface with GeckoView?

**Comment by Sdaswani** [comment]
@mmccorks can you please respond?

**Comment by mmccorks** [comment]
We just met to discuss this - we plan to meet again to discuss whether we should use the existing crash pings or send something custom (and what that should look like). But we agreed that we should send some sort of crash ping so that a) we can attribute crashes to a given build and b) consolidate native and managed crash counts in one place for better product decision making. Happy to chat with you to get your input @mcomella, this is an ongoing discussion.

**Comment by Baron-Severin** [comment]
Copying in some information from an email chain.

> I spent some time looking into this yesterday.
> 
> Top level conclusions:
> - Focus did not implement crash pings (work is not trivial, will get it from Glean). They use the A-C crash reporter and send native GV crashes to Socorro
> - They compare these to native crashes on Play store (**Do we have an equivelent way to track WV crashes?  If we're missing some, our comparison would be off.**)
> - A-C setup is very reasonable, and we can borrow from Focus
> 
> Reading through Fennec's crash ping implementation, it looks fairly hairy (https://bugzilla.mozilla.org/attachment.cgi?id=8945760&action=edit) so I would strongly prefer to wait for Glean integration to add this.  Focus dealt with the problem by comparing Play store native crash reports to Socorro numbers on GV problems.  This leads to the question, how are we tracking native system crashes?  Are we?  Presumably most of these are surfaced as exceptions, but we don't really have any way of being sure if they aren't being caught anywhere.  This might be something to bring up with Amazon.
> 
> Borrowing from the Focus implementation looks pretty straightforward, at least on the client (I'm not sure what goes into getting us set up with Socorro).  The A-C docs on the crash component (https://github.com/mozilla-mobile/android-components/blob/master/components/lib/crash/README.md) also seem very good, so I think we could get this running without too much effort.
> 
> 
> TL;DR: If we're currently tracking native WV crashes, it looks like we're in a good position to copy the Focus implementation.  If not, we need to find out how to handle that first.

**Comment by Baron-Severin** [comment]
@devinreams would you mind reaching out to Amazon and asking what they do with native WebView crashes?  They don't provide a dashboard, but even if we just had a total crash count to compare with GV, that would put us in a better position.

Adding @cpeterso for context.

**Comment by cpeterso** [comment]
Do we send FFTV+WebView crash reports to Sentry? We might have some crash stats from Sentry for comparison.

**Comment by liuche** [comment]
> Do we send FFTV+WebView crash reports to Sentry? We might have some crash stats from Sentry for comparison.

We do receive some crash reports (@psymoon has seen some), but we don't think that's all of them, and a partial count of crashes won't be as helpful of a benchmark for comparing the GV crash rate.

**Comment by psymoon** [comment]
@cpeterso @liuche I looked into it and I think we only get WebView crashes that _we have caused_

**Comment by mcomella** [comment]
@Baron-Severin Please unassign if this bug is inactive.

**Comment by Baron-Severin** [comment]
I don't believe this is actively being worked on.

**Comment by mcomella** [comment]
Closing in favor of implementation with Glean: #1932.
